### PR TITLE
Bridging the computer skill gap

### PR DESCRIPTION
**Pull request by Machminotaur**
#### Summary
Balance "Two Computers Books Added"

#### Purpose of change
Adds two new skillbooks to the Computer skill and associated droptables, removes a spurious S from an id of one of the books, and tightens one of the skill ranges on one of the other books. This should make getting to a usable skill level easier in the Computer skill, rather than being stopped at 2 because there was only one book capable of getting you to 3.

#### Describe the solution
Adds two books, makes "manual_computers" into "manual_computer" as well as every reference to it to keep it in-line with the other computer ids, and makes the advanced computing book a skill range of 4-6 rather than the 3-6 it was before.

#### Describe alternatives you've considered
There could be larger ranges but longer read times or more brutal enjoyment maluses, but I think the two new books on their own shore up the gap into the midrange books, and that is outside the purpose of this PR.

#### Testing
Game launched without error, teleported to a couple libraries and found both after a bit of searching.

### CODE REVIEW

**Comment by eltank** [comment]
Please don't change item IDs, you will break existing saves. If you really need to do that you have to write migration rules.

**Comment by actual-nh** [comment]
> Please don't change item IDs, you will break existing saves. If you really need to do that you have to also write migration rules converting the old ID to the new ID.

Good point; sorry, I forgot about that.

**Comment by eltank** [comment]
By adding more computer books to the loot tables you're also making computer books more likely to spawn, which I don't think is realistic (sadly). You should probably lower the weights of the new computer books and the existing ones that are similar in level.

**Comment by actual-nh** [comment]
> By adding more computer books to the loot tables without changing any weights you're also making computer books more likely to spawn, which I don't think is realistic (sadly). You should probably lower the weights for the new computer books as well as for the existing ones that are similar in level.

The existing ones spawn rather rarely; it is possible that they spawn unrealistically rarely. If so, I would not reduce the weights of any that are likely to be common (not requiring a very high skill or Int level).

**Comment by Machminotaur** [comment]
> You should probably lower the weights for the new computer books as well as for the existing ones that are similar in level.

I actually ended up cutting the largest chance for the manual in half because it was in the school drop table twice, at 20 twice. The drop rates for each book are at twenty combined at most in most locations (exceptions being the library specific shelves), and I flat out forgot to include all this because I had to rewrite the pull synopsis several times. The drop rates should look rather reasonable compared to other items in the list however.

The id change I'm not particularly married to, so that can be dropped.

**Comment by eltank** [comment]
The migration rule looks correct.

File: data/json/items/book/computer.json [hunk 1]
```diff
@@ -111,10 +111,52 @@
...
+    "max_level": 3,
+    "intelligence": 7,
+    "time": "20 m",
+    "fun": -1
+  },
+  {
+    "id": "hackerman_computer",
+    "type": "BOOK",
+    "name": { "str": "Pro Hacker Secrets", "str_pl": "copies of Pro Hacker Secrets" },
+    "//": "A thinly veiled attempt at teaching kids coding skills. Not nearly as exciting as the authors might have hoped, but there's some real knowledge in here.",
```
Comment (wapcaplet) [inline data/json/items/book/computer.json:145]: I think some or all of the descriptive text you have added to the comments for these books would be suitable in the description proper - at least in this case, I prefer the more colorful language of the comment.

Comment (Machminotaur) [reply data/json/items/book/computer.json]: Great! I think I originally meant to do it the other way around, but this worked as a way of seeing if people approved of the language too.

**Pull request merged**